modules/core/src/main/java/org/locationtech/jts/io/WKBReader.java

```diff
@@ -244,6 +245,14 @@ else if(isStrict)
     boolean hasM = ((typeInt & 0x40000000) != 0 || (typeInt & 0xffff)/1000 == 2 || (typeInt & 0xffff)/1000 == 3);
     //System.out.println(typeInt + " - " + geometryType + " - hasZ:" + hasZ);
     inputDimension = 2 + (hasZ ? 1 : 0) + (hasM ? 1 : 0);
+    
+    EnumSet<Ordinate> ordinateFlags = EnumSet.of(Ordinate.X, Ordinate.Y);
```

Comment:
Why does `WKBReader` need to change?  It already handles XYZM, I think?

Reply:
It made a check if the coordinates has a M coordinate but it doesn't consider it afterwards.
Now the create method of the csFactory is called accordingly.

Reply:
If this is fixing an issue with WKBReader as well, the PR title should reflect that.

Reply:
You're right, I changed the title.